src/web/utils/treeUtils.ts

```diff
@@ -43,16 +43,22 @@ function addNodeToTree(element: HTMLMarkdownElement, parentTreeNode: TreeNode, t
 
 function updateTreeElementRefs(treeRoot: TreeNode, element: HTMLMarkdownElement) {
   const stack: TreeNode[] = [treeRoot];
+  const treeElements = element.querySelectorAll('[data-id], [data-id] *') as NodeListOf<HTMLMarkdownElement>;
+  const treeElementsMapByDataID: Record<string, HTMLMarkdownElement> = {};
+  treeElements.forEach((el) => {
+    if (!el.getAttribute('data-id')) {
+      return;
+    }
+    treeElementsMapByDataID[el.getAttribute('data-id') || ''] = el;
+  });
```

Comment:
How about changing `treeElementsMapByDataID` name to `treeElementsMap` or `dataIDToElementMap`?